pallene/coder.lua

```diff
@@ -800,10 +800,12 @@ end
 
 function RecordCoder:primitive_slot(udata, field_name)
     local out = util.render(
-        [[((${STRUCT_NAME} *)getudatamem(${UDATA}))->${FIELD_NAME}]],
+        [[((${STRUCT_NAME} *)(cast_charp(${UDATA}) +
+            udatamemoffset(${NUVALUE})))->${FIELD_NAME}]],
```

Comment:
The pretty.lua formatter unfortunately removes this helpful indentation from the generated C file.